exporters/tests/test_failure_exporter.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+from jira import JIRA
+from jira.exceptions import JIRAError
+from prometheus_client.core import REGISTRY
+
+from failure.app import TrackerFactory
+from failure.collector_jira import JiraFailureCollector
+
+
+def setup_jira_collector(server, username, apikey) -> JIRA:
+    tracker_provider = "jira"
+
+    projects = None
+
+    jira_collector = TrackerFactory.getCollector(
+        username, apikey, server, projects, tracker_provider
+    )
+
+    return jira_collector
+
+
+@pytest.mark.parametrize(
+    "server, username, apikey",
+    [
+        (
+            "https://pelorustest.atlassian.net",
+            "fake@user.com",
+            "WIEds4uZHiCGnrtmgQPn9E7D",
+        )
+    ],
+)
+@pytest.mark.integration
+def test_jira_connection(server, username, apikey):
+
+    collector = setup_jira_collector(server, username, apikey)
+
+    with pytest.raises(JIRAError) as context_ex:
+        collector._connect_to_jira()
+
+    assert (
+        "You are not authenticated. Authentication required to perform this operation."
+        in str(context_ex.value)
+    )
+
+
+@pytest.mark.parametrize(
+    "server, username, apikey",
+    [("https://pelorustest.atlassian.net", "fake@user.com", "fakepass")],
+)
+@pytest.mark.integration
+def test_jira_pass_connection(server, username, apikey):
+
+    collector = setup_jira_collector(server, username, apikey)
+
+    with pytest.raises(JIRAError) as context_ex:
+        collector._connect_to_jira()
+
+    assert "Basic authentication with passwords is deprecated" in str(context_ex.value)
+
+
+@pytest.mark.parametrize(
+    "server, username, apikey",
+    [
+        (
+            "https://pelorustest.atlassian.net",
+            "fake@user.com",
+            "WIEds4uZHiCGnrtmgQPn9E7D",
+        )
+    ],
+)
+def test_jira_prometheus_register(server, username, apikey):
+    def mock_search_issues(self):
+        return []
+
+    monkeypatch = MonkeyPatch()
```

Comment:
The preferred way to use monkeypatch is as a fixture-- we can just make a parameter named `monkeypatch` and it will work.

Reply:
Yeah, I had troubles with fixture here, but will play with it.